src/Api/Controllers/OrganizationUsersController.cs

```diff
@@ -233,7 +243,24 @@ public async Task PutResetPasswordEnrollment(string orgId, string userId, [FromB
         [HttpPut("{id}/reset-password")]
         public async Task PutResetPassword(string orgId, string id, [FromBody]OrganizationUserResetPasswordRequestModel model)
         {
+            
             var orgGuidId = new Guid(orgId);
+            
+            // Org must be able to use reset password
+            var org = await _organizationRepository.GetByIdAsync(orgGuidId);
+            if (org == null || !org.UseResetPassword)
+            {
+                throw new BadRequestException("Organization does not allow password reset.");
+            }
+            
+            // Enterprise policy must be enabled 
+            var resetPasswordPolicy =
+                await _policyRepository.GetByOrganizationIdTypeAsync(orgGuidId, PolicyType.ResetPassword);
+            if (resetPasswordPolicy == null || !resetPasswordPolicy.Enabled)
+            {
+                throw new BadRequestException("Organization does not have the password reset policy enabled.");
+            }
```

Comment:
This is too much business logic in the controller. This should be shifted into a service method.

Reply:
Discussed. Will update.

Reply:
Updated.